### ISSUE TITLE
Add SPI for customizing metric export

### ISSUE THREAD

**Issue by jack-berg** [open-telemetry/opentelemetry-java#3437]
**Is your feature request related to a problem? Please describe.**
There is no way to configure custom metric exporters. 

`SdkMeterProviderConfigurer` provides a hook to register views and set the resource via `configure(SdkMeterProviderBuilder meterProviderBuilder)`. However, this is insufficient for customizing metric exporter configuration since both push (`IntervalMetricReader`) and pull (`PrometheusCollector`) based exporters require access to a built `SdkMeterProvider`. If you were to call `SdkMeterProviderBuilder#build()` in `configure(SdkMeterProviderBuilder)`, you'd have access to a different instance the one that gets registered globally.

**Describe the solution you'd like**
I'd like to extend `SdkMeterProviderConfigurer` with an additional callback method which would be invoked after the `SdkMeterProvider` has been built and registered globally. This would provide a place to customize metric exporters and `IntervalMetricReader` in ways not exposed by autoconfigure env vars / system properties.
```
  /** Callback which is invoked after the {@link SdkMeterProvider} has been built and registered globally. */
  default void callback(SdkMeterProvider sdkMeterProvider) {}
```

**Describe alternatives you've considered**
Also considered adding `ConfigurableMetricExporterProvider` akin to `ConfigureableSpanExporterProvider`, but doesn't solve the problem because metric exporters also need an `IntervalMetricReader` (or in prometheus' case a server to expose the data). Unclear how that would be exposed via SPI and tie in neatly with `ConfigurableMetricExporterProvider`. 

**Additional context**
I'm trying to use SPI hooks to customize `OtlpGrpcMetricExporter` to set my own `MangedChannel` which is configured with retry logic in accordance with the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#retry) but can't. I opened Issue #3271 to separately address adding retry logic to the OTLP exporters.

**Comment by jkwatson** [comment]
Yes, we had been holding off on this until the metrics SDK got more stable.
